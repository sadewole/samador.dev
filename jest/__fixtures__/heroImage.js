module.exports = {
  data: {
    mobileImage: {
      childImageSharp: {
        fixed: {
          base64: "test",
          height: 200,
          src: "photo.jpg",
          srcSet: "photo.jpg",
          width: 500,
        },
      },
    },
    desktopImage: {
      childImageSharp: {
        fixed: {
          base64: "test",
          height: 200,
          src: "photo1.jpg",
          srcSet: "photo1.jpg",
          width: 500,
        },
      },
    },
  },
}
