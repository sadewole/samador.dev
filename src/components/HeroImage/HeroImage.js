import React from "react"
import PropTypes from "prop-types"
import Img from "gatsby-image"

const Image = ({ data }) => {
  const sources = [
    data.mobileImage.childImageSharp.fixed,
    {
      ...data.desktopImage.childImageSharp.fixed,
      media: `(min-width: 800px)`,
    },
  ]

  return (
    <Img
      fixed={sources}
      objectFit="cover"
      objectPosition="50% 50%"
      className="bg-gray-800 absolute -top-225-px -right-225-px rounded-full opacity-80 hidden md:block"
    />
  )
}

Image.propTypes = {
  data: PropTypes.object.isRequired,
}

export default Image
