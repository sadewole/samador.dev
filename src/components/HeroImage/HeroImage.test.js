import React from "react"
import renderer from "react-test-renderer"

import HeroImage from "./HeroImage"
import heroImage from "../../../jest/__fixtures__/heroImage"

describe("HeroImage", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      value: () => ({
        matches: false,
        addListener: () => {},
        removeListener: () => {},
      }),
    })
  })

  test("render correctly", () => {
    const tree = renderer.create(<HeroImage {...heroImage} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
