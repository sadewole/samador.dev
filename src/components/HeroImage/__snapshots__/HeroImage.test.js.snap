// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroImage render correctly 1`] = `
<div
  className="bg-gray-800 absolute -top-225-px -right-225-px rounded-full opacity-80 hidden md:block gatsby-image-wrapper"
  style={
    Object {
      "display": "inline-block",
      "height": 200,
      "overflow": "hidden",
      "position": "relative",
      "width": 500,
    }
  }
>
  <picture>
    <source
      media="(min-width: 800px)"
      srcSet="test"
    />
    <source
      srcSet="test"
    />
    <img
      alt=""
      aria-hidden={true}
      src="test"
      style={
        Object {
          "height": "100%",
          "left": 0,
          "objectFit": "cover",
          "objectPosition": "center",
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "transitionDelay": "500ms",
          "width": "100%",
        }
      }
    />
  </picture>
  <picture>
    <source
      media="(min-width: 800px)"
      srcSet="photo1.jpg"
    />
    <source
      srcSet="photo.jpg"
    />
    <img
      alt=""
      height={200}
      loading="lazy"
      onLoad={[Function]}
      src="photo.jpg"
      srcSet="photo.jpg"
      style={
        Object {
          "height": "100%",
          "left": 0,
          "objectFit": "cover",
          "objectPosition": "center",
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transition": "opacity 500ms",
          "width": "100%",
        }
      }
      width={500}
    />
  </picture>
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "<picture><source media=\\"(min-width: 800px)\\" srcset=\\"photo1.jpg\\" /><source srcset=\\"photo.jpg\\" /><img loading=\\"lazy\\" width=\\"500\\" height=\\"200\\" srcset=\\"photo.jpg\\" src=\\"photo.jpg\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
      }
    }
  />
</div>
`;
